### PR TITLE
[NEW] Log events from Onboarding, NewServer, Login and Register screens

### PR DESCRIPTION
**Pull request by youssef-md**
### What was done in this PR:
* Creation of `logEvent` method to isolate to logic to improve future maintenance.
*  File `/utils/log/events.js` to define all the current events, to easily find them and to prevent typo if an event would be used in multiple places.

#### OnboardingView: 
* Join a workspace.
* Create new Workspace(with fail event).

#### NewServerView:
* Connect to typed workspace URL(with fail event).
* Join the open workspace.

#### LoginServices:
* Log all the coded OAuth options.

#### LoginView:
* Log the default login(with fail event).

#### RegisterView:
* Log the default sign up(with fail event).

Since Firebase Analytics Events can take a while to appear to the dashboard, I'll be using the [Analytics Debug View](https://firebase.google.com/docs/analytics/debugview) to test the event sending.

Here is a screenshot of Analytics Debug View with all the tracked events in this PR:
Screenshot from 2020-06-08 00-38-57

**user_engagement** and **screen_view** are default Firebase events, in this screenshot I selected the **join_open_workspace** event to show all the default payloads, the best one is **firebase_screen** which references the screen that the selected event was triggered.

### CODE REVIEW

File: app/utils/log.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ export const { leaveBreadcrumb } = bugsnag;
 
 let metadata = {};
 
+export const trackUserEvent = ({ eventName, payload }) => {
```
Comment (diegolmello) [inline app/utils/log.js:13]: Follow `logEvent`'s signature in order to remove clutter: https://v5.rnfirebase.io/docs/v5.x.x/analytics/reference/analytics#logEvent

File: app/utils/log.js [hunk 2]
```diff
@@ -10,6 +10,10 @@ export const { leaveBreadcrumb } = bugsnag;
 
 let metadata = {};
 
+export const trackUserEvent = (eventName, payload) => {
```
Comment (diegolmello) [inline app/utils/log.js:13]: Rename to `logEvent`

File: app/utils/trackableEvents.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export const JOIN_A_WORKSPACE = 'join_a_workspace';
```
Comment (diegolmello) [inline app/utils/trackableEvents.js:1]: Create a `log` folder under `app/utils` and move this file and `log.js` as well.

Comment (diegolmello) [inline app/utils/trackableEvents.js:1]: Rename this file `events.js`

Comment (diegolmello) [inline app/utils/trackableEvents.js:1]: Create a file `app/utils/log/index.js` with all exports.
This way you can import `logEvent` and `events` from the same source.

Comment (youssef-md) [reply app/utils/trackableEvents.js:1]: I've placed all the `log.js` code inside the `log/index.js` together with the export for the events.

**Pull request merged**